kernel/path/src/lib.rs

```diff
@@ -130,11 +130,11 @@ impl Path {
         let mut first_cmpnt = true; 
         for component in new_components {
             if first_cmpnt {
-                new_path.push_str(&format!("{}",  component));
+                new_path.push_str(component);
                 first_cmpnt = false;
             } 
             else {
-                new_path.push_str(&format!("/{}",  component));
```

Comment:
This seems like an oversight, you probably changed the above and thought this was doing the same, but it actually adds a `/` in front.